libexec/helpers/azure-machine-type-definitions.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+#==============================================================================
+# Copyright (C) 2020-present Alces Flight Ltd.
+#
+# This file is part of Flight Action API.
+#
+# This program and the accompanying materials are made available under
+# the terms of the Eclipse Public License 2.0 which is available at
+# <https://www.eclipse.org/legal/epl-2.0>, or alternative license
+# terms made available by Alces Flight Ltd - please direct inquiries
+# about licensing to licensing@alces-flight.com.
+#
+# Flight Action API is distributed in the hope that it will be useful, but
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, EITHER EXPRESS OR
+# IMPLIED INCLUDING, WITHOUT LIMITATION, ANY WARRANTIES OR CONDITIONS
+# OF TITLE, NON-INFRINGEMENT, MERCHANTABILITY OR FITNESS FOR A
+# PARTICULAR PURPOSE. See the Eclipse Public License 2.0 for more
+# details.
+#
+# You should have received a copy of the Eclipse Public License 2.0
+# along with Flight Action API. If not, see:
+#
+#  https://opensource.org/licenses/EPL-2.0
+#
+# For more information on Flight Action API, please visit:
+# https://github.com/openflighthpc/flight-action-api
+#===============================================================================
+
+# A map from generic machine types to the equivalent Azure instance type.
+#
+# NOTE: If changing this map also change the types listed in the estate
+# metadata files.
+declare -A MACHINE_TYPE_MAP
+MACHINE_TYPE_MAP=(
+  [general-small]=Standard_B2s
+  #[general-medium]=
+  #[general-large]=
```

Comment:
@ColonelPanicks you're cost calculator doesn't have instance types defined for azure for `general-medium`, `general-large` or `gpu-large`.  As things stand at the moment this means that the help text will not be correct.  That is, the help text will claim `flight estate change azure-node01 general-medium` is valid, but trying to run that will result in an error that `general-medium` is not a valid machine type.

We have a few solutions:

1. Modify the cost calculator spreadsheet to specify azure types for these machine types.
2. Adjust the help text in some way for azure nodes.  This is unlikely to work very well as running `flight estate change --help` provides no indication of what type of node it is for.
3. Adjust the error message so that we distinguish between invalid machine types, such as `general-lrage`, and valid but unsupported machine types, such as `general-large` for azure nodes.

My favourite option here is (1).

Reply:
My favourite option is also number 1, but alas, it's not going to be that simple. There are suitable candidates for `general-small` and `general-large` and these shall be dealt with separately. However the largest azure gpu node is `Standard_NC24s_v3`, so `gpu-large` can't be supported.

My proposal is a combination of options 2 and 3. Put a "not supported on all platforms" note in the help text, and distinguish the error in the script.

Reply:
With help from @WilliamMcCumstie the general-medium and general-large types have been sorted for Azure. The only outstanding issue is the gpu-large, the reason there isn't one for this is that there currently isn't an Azure instance type that boasts 8 x v100 GPUs like on AWS in the region we default to (`uksouth`).

From the looks of things, the `ND40rs v2` is the closest Azure instance type to the `p3.16xlarge` although it does cost slightly less per month. I'll add this one into my sheet as, at the very least, an indicator of the similar instance type in Azure.

Reply:
Thanks for the response.  If you're adding that to your sheet, I'll go ahead and make the change to `flight estate change node01 gpu-large` will use `ND40rs v2` on azure.

Reply:
Can you also add in the missing machine types.  The cloud cost spreadsheet has entries for all of azure now.